new-lamassu-admin/src/pages/Machines/Machines.js

```diff
@@ -38,32 +35,34 @@ const GET_INFO = gql`
         label
         type
       }
+      latestEvent {
+        note
+      }
     }
     config
   }
 `
 
-const getMachines = R.path(['machines'])
+const getMachineID = path => path.slice(path.lastIndexOf('/') + 1)
 
 const Machines = () => {
-  const { data, refetch, loading } = useQuery(GET_INFO)
   const location = useLocation()
-  const [selectedMachine, setSelectedMachine] = useState('')
+  const [machine, setMachine] = useState({})
+  const [config, setConfig] = useState({})
+  const { data, refetch } = useQuery(GET_INFO, {
+    variables: {
+      deviceId: getMachineID(location.pathname)
+    },
+    onCompleted: () => {
+      setMachine(data.machine)
+      setConfig(data.config)
+    }
+  })
```

Comment:
I don't think we need useState anymore
```suggestion
  const { data, refetch } = useQuery(GET_INFO, {
    variables: {
      deviceId: getMachineID(location.pathname)
    }
  })
  
  const machine = R.path(['machine'])(data) ?? {}
  const config = ...
```

Reply:
Nice call on this, completely agree with you Cesar. 
At the time, I thought I actually needed this to handle the changes from the 'Action buttons' on the page and was having trouble with this. 

Thanks for the feedback!

Reply:
no problem, glad to help!